pkg/daemon/writer.go

```diff
@@ -19,55 +27,106 @@ const (
 	machineConfigDaemonSSHAccessAnnotationKey = "machineconfiguration.openshift.io/ssh"
 	// MachineConfigDaemonSSHAccessValue is the annotation value applied when ssh access is detected
 	machineConfigDaemonSSHAccessValue = "accessed"
+
+	nodeWriterKubeconfigPath = "/var/lib/kubelet/kubeconfig"
 )
 
 // message wraps a client and responseChannel
 type message struct {
-	client          corev1client.NodeInterface
-	lister          corev1lister.NodeLister
-	node            string
 	annos           map[string]string
 	responseChannel chan error
 }
 
 // clusterNodeWriter is a single writer to Kubernetes to prevent race conditions
 type clusterNodeWriter struct {
-	writer chan message
+	nodeName         string
+	writer           chan message
+	client           corev1client.NodeInterface
+	lister           corev1lister.NodeLister
+	nodeListerSynced cache.InformerSynced
+	kubeClient       kubernetes.Interface
 }
 
 // NodeWriter is the interface to implement a single writer to Kubernetes to prevent race conditions
 type NodeWriter interface {
 	Run(stop <-chan struct{})
-	SetDone(client corev1client.NodeInterface, lister corev1lister.NodeLister, node string, dcAnnotation string) error
-	SetWorking(client corev1client.NodeInterface, lister corev1lister.NodeLister, node string) error
-	SetUnreconcilable(err error, client corev1client.NodeInterface, lister corev1lister.NodeLister, node string) error
-	SetDegraded(err error, client corev1client.NodeInterface, lister corev1lister.NodeLister, node string) error
-	SetSSHAccessed(client corev1client.NodeInterface, lister corev1lister.NodeLister, node string) error
+	SetDone(dcAnnotation string) error
+	SetWorking() error
+	SetUnreconcilable(err error) error
+	SetDegraded(err error) error
+	SetSSHAccessed() error
+	SetAnnotations(annos map[string]string) error
+}
+
+func (nw *clusterNodeWriter) handleNodeWriterEvent(node interface{}) {
+	n := node.(*corev1.Node)
+
+	if n.Name != nw.nodeName {
+		return
+	}
+
+	glog.V(4).Infof("Updating Node %s", n.Name)
```

Comment:
Currently this is just dead code - I think we probably should change the daemon to use this to *read* the node too but it's a more invasive change than I want to try to land as one unit, and not on the critical path (see https://github.com/openshift/machine-config-operator/pull/3143#discussion_r870503417 )

What took me a while to debug is that the informer won't actually return "I started" unless you attach an event handler.

I'm only creating an informer here because `internal.UpdateNodeRetry` wants one, but I could change that to support doing a direct client `Get()`.

OTOH...hmmm...we may suffer from races unless the daemon and nodeWriter actually share a common lister state for the node object.

Going to plow forward with just "phase 0" but if we find races we may need to go to "phase 1".